broker/src/tunneldigger_broker/hooks.py

```diff
@@ -55,6 +55,10 @@ def close(self):
         Closes the hook process.
         """
 
+        if not hasattr(self, "buffer"):
```

Comment:
I think it would be better to have an explicit `is_closed` attribute on the object so that we can immediately set it the first thing in `close`. Otherwise there might be some race conditions.

Reply:
That's still not atomic, so it will not solve race conditions. But also, this is all sequential, right?

Reply:
Probably true.